### ISSUE TITLE
Menu: keyboard interaction and discoverability

### ISSUE THREAD

**Issue by dfarough** [wet-boew/wet-boew#7537]
@wet-boew/access-working-group 
@thomasgohard 
Can we improve usability for these menus by providing more information on how to use these menus for Keyboard only and screen reader users? 

IF I have Jaws configured to provide menu and control help information, Jaws will say "To move through items press up or down arrow." when the first menu item in the menu bar gains focus.  If however, the first item in the menu bar does not contain any menu items, using up or down arrow does nothing.  For these cases, it would be helpful to include instructions for using the left and right arrows as well.

At the same time can we also include some visual queues for keyboard only users either through CSS or some other mechanism so that users don't get confused when trying to move horizontally across the menu bar.  Users may get frustrated if after focussing on the first menu item they press tab and leave the menu structure.

I think we need to provide more information for new users of these menus.  This would also be a good thing for the tabbed interface and carousel.

**Comment by dfarough** [comment]
Maybe something like the following would be good text for screen reader users.
"Use Arrow keys to move vertically or horizontally through the menus."

**Comment by mysticjeff** [comment]
This makes sense to me too 

-----Original Message-----

**Comment by dfarough** [comment]
Could someone please address this issue by modifying the text presented when these menus get focus.  Our application developers are waiting on a fix for this because the Public Service Recruiting system contains a menu structure where the first 3 menu items on the menubar do not contain any sub menus.  As a result, when the home menu gains focus a Jaws user is told to move through items press up or down arrow.  In the case of this menu structure, pressing up or down arrow does nothing when Jaws is in focus mode.  In this case 
"Use Arrow keys to move vertically or horizontally through the menus." would be a better choice.   Particularly since there is no way to communicate whether a menu bar is vertical or horizontal.

At present users of NVDA do not even get this much information.  I will add an issue to the github repository for NVDA requesting this so that users will get more information when encountering Menus and tab lists ETC.

 




































































"Use Arrow keys to move vertically or horizontally through the menus."

**Comment by dfarough** [comment]
I added an issue to the NVDA resource on Github and found out that there is an NVDA  plug-in that will provide contextual tutor information for the control with focus.  The plug-in is called Control usage assistant.  the user can press the NVDA key in conjunction with "H" to hear the desired information. It's too bad that new users of NVDA will need to download an ad-Din to get this information, but it seems unlikely that NV Access will add this capability to NVDA.

**Comment by hsrudnicki** [comment]
Does this remain still an issue or otherwise? Either way, please update the status of this again, whether this can be closed or otherwise, thank you

**Comment by juleskuehn** [comment]
This is still an issue. Without being able to see the screen, it is unclear whether to use the left+right or up+down arrow keys to descend into submenus or to move between menu items at the top level.

"Use left and right arrow keys to navigate through top-level menu items. Up and down arrow keys navigate through sub-menu items"

**Comment by juleskuehn** [comment]
Along with the other issues I referenced today, this ultimately appears to be a duplicate of Issue #6682, which means there are 4 open issues requesting instructional text for keyboard navigation of menus.
